File: lib/features/community/domain/community_post.dart
import '../../daily/domain/donation_event.dart';

class CommunityPost {
  const CommunityPost({
    required this.id,
    required this.slug,
    required this.title,
    required this.content,
    required this.audienceLabel,
    required this.viewsCount,
    required this.sharesCount,
    this.excerpt,
    this.imageUrl,
    this.publishedAt,
    this.province,
  });

  factory CommunityPost.fromJson(Map<String, dynamic> json) {
    return CommunityPost(
      id: json['id'] as String,
      slug: json['slug'] as String,
      title: json['title'] as String,
      excerpt: json['excerpt'] as String?,
      content: json['content'] as String? ?? '',
      imageUrl: json['image_url'] as String?,
      publishedAt: json['published_at'] == null
          ? null
          : DateTime.parse(json['published_at'] as String),
      audienceLabel: json['audience_label'] as String? ?? 'Tất cả người dùng',
      province: json['province'] is Map<String, dynamic>
          ? AdministrativeArea.fromJson(json['province'] as Map<String, dynamic>)
          : null,
      viewsCount: json['views_count'] as int? ?? 0,
      sharesCount: json['shares_count'] as int? ?? 0,
    );
  }

  final String id;
  final String slug;
  final String title;
  final String? excerpt;
  final String content;
  final String? imageUrl;
  final DateTime? publishedAt;
  final String audienceLabel;
  final AdministrativeArea? province;
  final int viewsCount;
  final int sharesCount;
}

File: backend/app/Models/CommunityPost.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CommunityPost extends Model
{
    use HasFactory;

    protected $fillable = [
        'hospital_id',
        'author_id',
        'title',
        'slug',
        'excerpt',
        'content',
        'image_url',
        'status',
        'published_at',
        'audience_type',
        'target_blood_type',
        'target_hero_level',
        'province_code',
        'ward_code',
        'views_count',
        'shares_count',
    ];

    protected function casts(): array
    {
        return [
            'published_at' => 'datetime',
        ];
    }

    public function hospital()
    {
        return $this->belongsTo(Hospital::class);
    }

    public function author()
    {
        return $this->belongsTo(User::class, 'author_id');
    }

    public function province()
    {
        return $this->belongsTo(Province::class, 'province_code', 'code');
    }

    public function ward()
    {
        return $this->belongsTo(Ward::class, 'ward_code', 'code');
    }
}
